swarm/api/http/server_test.go

```diff
@@ -18,21 +18,113 @@ package http_test
 
 import (
 	"bytes"
+	"crypto/rand"
 	"errors"
 	"fmt"
 	"io/ioutil"
 	"net/http"
+	"os"
 	"strings"
 	"sync"
 	"testing"
 
 	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
 	"github.com/ethereum/go-ethereum/swarm/api"
 	swarm "github.com/ethereum/go-ethereum/swarm/api/client"
 	"github.com/ethereum/go-ethereum/swarm/storage"
 	"github.com/ethereum/go-ethereum/swarm/testutil"
 )
 
+func init() {
+	log.Root().SetHandler(log.CallerFileHandler(log.LvlFilterHandler(log.LvlTrace, log.StreamHandler(os.Stderr, log.TerminalFormat(true)))))
+}
+
+// \TODO if create -> get -> update -> get, the last get with return 1.1 because 1.2 retrieve is still pending
+func TestBzzResource(t *testing.T) {
+	srv := testutil.NewTestSwarmServer(t)
+	defer srv.Close()
+
+	// our mutable resource "name"
+	keybytes := make([]byte, common.HashLength)
+	copy(keybytes, []byte{42})
+	srv.Hasher.Reset()
+	srv.Hasher.Write([]byte(fmt.Sprintf("%x", keybytes)))
```

Comment:
Is it correct to be writing the hex representation of the key to the hasher, rather than the bytes themselves?

For example, I'd expect this to be:

```go
srv.Hasher.Write(keybytes)
```

Reply:
It's correct because we're getting it in hex from the HTTP server.

Reply:
I'm not sure I follow. So you have `keybytes` which is the "key" of the resource. Why would I want to create a cryptographic hash of the hex representation of `keybytes` rather than just the hash of `keybytes`? It feels like a waste of time (the hex representation is twice as many bytes)?

Reply:
Ah, I this understand the confusion. These first "keybytes" are merely the "name" of the resource. The resource name is always hashed, and the hash is the key of the root chunk. In the backend, as you see, there are no double hashes. But in the test we need to to make the data match.